src/main/kotlin/com/jetbrains/micropython/nova/actions.kt

```diff
@@ -170,8 +170,12 @@ class InstantRun : DumbAwareAction() {
     override fun getActionUpdateThread(): ActionUpdateThread = BGT
 
     override fun update(e: AnActionEvent) {
-        e.presentation.isEnabled = e.getData(CommonDataKeys.VIRTUAL_FILE) != null
-
+        val file = e.getData(CommonDataKeys.VIRTUAL_FILE)
+        val files = e.getData(CommonDataKeys.VIRTUAL_FILE_ARRAY)
+        e.presentation.isEnabled = file != null &&
+                !file.isDirectory &&
+                file.extension == "py" &&
```

Comment:
I don't see a point to prevent users from executing non-.py files.
BTW, the proper way to check the file type is `virtualFile.fileType == PythonFileType.INSTANCE`

Reply:
> I don't see a point to prevent users from executing non-.py files. BTW, the proper way to check the file type is `virtualFile.fileType == PythonFileType.INSTANCE`

1. I suppose you are right on this, I'll change it.
2. Thanks for letting me know!